routes/mentor.go

```diff
@@ -3,8 +3,45 @@ package routes
 import (
 	"fmt"
 	"net/http"
+	"encoding/json"
+	"io/ioutil"
+	"github.com/jinzhu/gorm"
+	"kwoc20-backend/models"
 )
 
 func MentorReg(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintf(w, "hello")
+
+	var mentor models.Mentor
+	body, _ := ioutil.ReadAll(r.Body)
+	err := json.Unmarshal(body, &mentor)
+	if err != nil {
+	  http.Error(w, err.Error(), 400)
+	  return 		
+	}
+
+	
+	db, err := gorm.Open("sqlite3", "kwoc.db")
+	if err != nil {
+		 http.Error(w, err.Error(), 500)
+		 return
+	}
+	defer db.Close()
+
+	err = db.Create(&models.Mentor{
```

Comment:
Why not save the `mentor` variable itself?

Reply:
Actually, I also did a similar thing in `project` routes, not saving the variable directly
my point in doing so was that we sometimes frontend may send some extra things, which aren't to be saved in DB(req for verification) may be
actually in this case, there wont be anything of such, we can do as u told

Reply:
If any extra data is sent, won't the unmarshalling automatically fail?
We have fixed the type by declaring `mentor` as `model.Mentor`.

Reply:
We can do something as such? Dynamic unmarshalling.
https://medium.com/@nate510/dynamic-json-umarshalling-in-go-88095561d6a0

Reply:
Article is nice! But since we are at an early stage in our project, let us not resort to ad-hoc designing such as that. You can safely save your data. If unmarshalling fails, just send a `http.StatusBadRequest`.
What data is required should be in the API docs (or comments atleast)

Reply:
Umm okay :)

Reply:
> If any extra data is sent, won't the unmarshalling automatically fail?
We have fixed the type by declaring mentor as model.Mentor.

provided we won't use interface {} type,

ya for now saving directly `mentor` is good option, since nothing more will be sent from frontend